src/lib.rs

```diff
@@ -76,28 +86,372 @@ impl ULinkZenoh {
             .replace("//", "\\/");
         Ok(zenoh_key)
     }
+
+    // TODO: We need a standard way in uprotocol-rust to change UUID to String
+    fn uuid_to_string(uuid: &Uuid) -> String {
+        format!("{}:{}", uuid.msb, uuid.lsb)
+    }
+
+    async fn send_publish(
+        &self,
+        zenoh_key: &str,
+        payload: UPayload,
+        attributes: UAttributes,
+    ) -> Result<(), UStatus> {
+        // Get the data from UPayload
+        let Some(Data::Value(buf)) = payload.data else {
+            // TODO: Assume we only have Value here, no reference for shared memory
+            return Err(UStatus::fail_with_code(
+                UCode::InvalidArgument,
+                "Invalid data",
+            ));
+        };
+
+        // Serialized UAttributes into protobuf
+        // TODO: Should we map priority into Zenoh priority?
+        let mut attr = vec![];
+        let Ok(()) = attributes.encode(&mut attr) else {
+            return Err(UStatus::fail_with_code(
+                UCode::InvalidArgument,
+                "Unable to encode UAttributes",
+            ));
+        };
+
+        // Add attachment and payload
+        let mut attachment = AttachmentBuilder::new();
+        attachment.insert("uattributes", attr.as_slice());
+        let putbuilder = self
+            .session
+            .put(zenoh_key, buf)
+            .encoding(Encoding::WithSuffix(
+                KnownEncoding::AppCustom,
+                payload.format.to_string().into(),
+            ))
+            .with_attachment(attachment.build());
+
+        // Send data
+        if putbuilder.res().await.is_err() {
```

Comment:
I suggest to use the more rusty `.map_err(|e| ...)?;`.
In general, if an error is returned, it's always good to report it up as part of the message.

Reply:
Fixed!